pkg/service/restore/restore_integration_test.go

```diff
@@ -334,3 +338,119 @@ func TestRestoreTablesVnodeToTabletsIntegration(t *testing.T) {
+	createTable(t, h.dstCluster.rootSession, ks1, tab)
+	tab1 := randomizedName("tab_1_")
+	createTable(t, h.srcCluster.rootSession, ks2, tab1)
+	createTable(t, h.dstCluster.rootSession, ks2, tab1)
+	tab2 := randomizedName("tab_2_")
+	createTable(t, h.srcCluster.rootSession, ks2, tab2)
+	createTable(t, h.dstCluster.rootSession, ks2, tab2)
+
+	Print("View setup")
+	mv := randomizedName("mv_")
+	CreateMaterializedView(t, h.srcCluster.rootSession, ks1, tab, mv)
+	CreateMaterializedView(t, h.dstCluster.rootSession, ks1, tab, mv)
+	si := randomizedName("si_")
+	CreateSecondaryIndex(t, h.srcCluster.rootSession, ks1, tab, si)
+	CreateSecondaryIndex(t, h.dstCluster.rootSession, ks1, tab, si)
+	mv1 := randomizedName("mv_1_")
+	CreateMaterializedView(t, h.srcCluster.rootSession, ks2, tab1, mv1)
+	CreateMaterializedView(t, h.dstCluster.rootSession, ks2, tab1, mv1)
+
+	Print("Fill setup")
+	fillTable(t, h.srcCluster.rootSession, 100, ks1, tab)
+	fillTable(t, h.srcCluster.rootSession, 100, ks2, tab1, tab2)
+
+	units := []backup.Unit{
+		{
+			Keyspace:  ks1,
+			Tables:    []string{tab, mv, si + "_index"},
+			AllTables: true,
+		},
+		{
+			Keyspace:  ks2,
+			Tables:    []string{tab1, tab2, mv1},
+			AllTables: true,
+		},
+	}
+
+	Print("Run backup")
+	loc := []Location{testLocation("paused", "")}
+	S3InitBucket(t, loc[0].Path)
+
+	// Starting from SM 3.3.1, SM does not allow to back up views,
+	// but backed up views should still be tested as older backups might
+	// contain them. That's why here we manually force backup target
+	// to contain the views.
+	ctx := context.Background()
+	h.srcCluster.TaskID = uuid.NewTime()
+	h.srcCluster.RunID = uuid.NewTime()
+
+	rawProps, err := json.Marshal(map[string]any{"location": loc})
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "marshal properties"))
+	}
+
+	target, err := h.srcBackupSvc.GetTarget(ctx, h.srcCluster.ClusterID, rawProps)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "generate target"))
+	}
+	target.Units = units
+
+	err = h.srcBackupSvc.Backup(ctx, h.srcCluster.ClusterID, h.srcCluster.TaskID, h.srcCluster.RunID, target)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "run backup"))
+	}
+
+	pr, err := h.srcBackupSvc.GetProgress(ctx, h.srcCluster.ClusterID, h.srcCluster.TaskID, h.srcCluster.RunID)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "get progress"))
+	}
+	tag := pr.SnapshotTag
+
+	Print("Run restore tables")
+	grantRestoreTablesPermissions(t, h.dstCluster.rootSession, []string{ks1, ks2}, h.dstUser)
+	props := map[string]any{
+		"location":       loc,
+		"keyspace":       []string{ks1, ks2},
+		"snapshot_tag":   tag,
+		"restore_tables": true,
+	}
+	err = runPausedRestore(t, func(ctx context.Context) error {
+		h.dstCluster.RunID = uuid.NewTime()
+		rawProps, err := json.Marshal(props)
+		if err != nil {
+			return err
+		}
+		return h.dstRestoreSvc.Restore(ctx, h.dstCluster.ClusterID, h.dstCluster.TaskID, h.dstCluster.RunID, rawProps)
+	}, 45*time.Second, 2)
```

Comment:
How do you assure that the restore is paused after the view is removed ?
Maybe I'm wrong, but it looks bit random to me.

Reply:
I don't - the general idea is that this test should pause restore in random places, so not only when we set up some hook via client interceptor. This might give flaky false positives, but it increases the general coverage.

But since the restore is paused after 45 seconds and views are dropped at the beginning of the restore, it's really unlikely that the views won't be dropped before the first pause.

Reply:
> I don't - the general idea is that this test should pause restore in random places, so not only when we set up some hook via client interceptor. 

But the origin for this change comes from the fact that the bug is found which root cause is:
- view is dropped 
- restore is paused
- restore is resumed
- Scylla Manager takes the SSTables of the view and restores them
Let the test to reproduce the issue explicitly.

> This might give flaky false positives, but it increases the general coverage.

We don't want flaky false positives.

We not to use interceptor on some calls or just use a test hook indicating that the view is drop ?

Reply:
I agree that we should explicitly test for pausing after views are dropped, but we somewhat need to introduce a flaky test pausing restore at random moments, as it improves coverage with places that we didn't think could cause problems with pause/resume. It won't cause CI to fail, but it might catch some issue from time to time.